### PR TITLE
Allow to execute a tox target without pinned dependencies

### PR DESCRIPTION
**Pull request by adferrand**
In response to #6582.

This PR passes the `CERTBOT_NO_PIN` environment variable to the unit tests tox envs. By setting `CERTBOT_NO_PIN` to `1` before executing a given tox env, certbot dependencies will be installed at their latest version instead of the usual pinned version.

I also moved the unpin logic one layer below to allow it to be used potentially more widely, and avoid unnecessary merging constraints operation in this case.

As warnings are errors now, latest versions of Python will break now the tests, because collections launch a warning when some classes are imported from `collections` instead of `collections.abc`. Certbot code is patched, and warning is ignored for now, because a lot of third party libraries still depend on this behavior.

An another PR will follow for `test-everything` branch.

### CODE REVIEW

**Comment by adferrand** [comment]
cc @bmw

**Comment by bmw** [comment]
I think my review on this above may have been missed.

**Comment by adferrand** [comment]
Sorry, did not catch it, I do it know.

File: tools/pip_install.py [hunk 1]
```diff
@@ -75,19 +75,29 @@ def main(args):
...
+            if requirements:
+                call_with_print('"{0}" -m pip install --constraint "{1}" --requirement "{2}"'
+                                .format(sys.executable, all_constraints, requirements))
 
-        merge_requirements(tools_path, test_constraints, all_constraints)
-        if requirements:
-            call_with_print('"{0}" -m pip install --constraint "{1}" --requirement "{2}"'
-                            .format(sys.executable, all_constraints, requirements))
+            with open(all_constraints, 'r') as file:
+                print(file.read())
```
[the remark this answers is not in the document]
Comment (adferrand) [reply tools/pip_install.py]: Yes, debug code. I remove it.

**Comment by adferrand** [comment]
I integrated the changes from #6595.

File: tox.ini [hunk 2]
```diff
@@ -64,6 +64,10 @@ source_paths =
     tests/lock_test.py
 
 [testenv]
+passenv = 
+    TRAVIS
+    APPVEYOR
```
Comment (bmw) [inline tox.ini]: The first two environment variables should no longer be needed here.

Comment (adferrand) [reply tox.ini]: Yes you are right. For the CI, either we are on travis and `export TOX_TESTENV_PASSENV=TRAVIS` applies, or we are on appveyor and `set TOX_TESTENV_PASSENV=APPVEYOR` applies.

And outside the CIs, `TRAVIS` and `APPVEYOR` are not set.

**Pull request merged**